recipe/meta.yaml

```diff
@@ -1,48 +1,85 @@
 {% set name = "pymilvus" %}
-{% set version = "1.1.2" %}
-
+{% set version = "2.4.5" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pymilvus-{{ version }}.tar.gz
-  sha256: 8622233c49aa9d27b84e6ee54f30d6370136a8ef4f2362b95d6499a21f5305c0
+  sha256: 1a497fe9b41d6bf62b1d5e1c412960922dde1598576fcbb8818040c8af11149f
+  patches:
+    - patches/0001-remove-milvus-lite-dependency.patch
+    - patches/0002-remove-grpc_testing-in-conftest.patch
 
 build:
   number: 0
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<38]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
+  build:
+    - patch     # [not win]
+    - m2-patch  # [win]
   host:
     - pip
-    - python >=3.6
-    - setuptools-scm
+    - python
+    - setuptools-scm >=6.2
+    - setuptools >69
+    - setuptools <70.1  # [py<39]
+    - wheel
     - gitpython
   run:
-    - grpcio >=1.22.0,<1.38.0
-    - grpcio-tools >=1.22.0,<1.38.0
-    - python >=3.6
-    - requests >=2.22.0
+    - python
+    - grpcio >=1.49.1,<=1.63.0
+    - protobuf >=3.20.0
+    - environs <=9.5.0
     - ujson >=2.0.0
-    - mmh3
+    - pandas >=1.2.4
+    - numpy <1.25.0  # [py<=38]
+  run_constrained:
+    # we release a version of pymilvus without
+    # the mandatory dependency to milvus-lite
+    # which we make optional
+    - milvus-lite >=2.4.0,<2.5.0  # [not win]
+    # setuptools is not a run dependency but
+    # it is pulled in by some other package
+    # and these constraints allow to respect
+    # the ones specified upstream in pyproject.toml
+    - setuptools >69
```

Comment:
`setuptools` is a required dependency because they import `pkg_resources`, see https://github.com/milvus-io/pymilvus/blob/42a66f14b109f59df39fa6907789f2514514b92c/pymilvus/client/__init__.py#L6. `pkg_resources` is bunbled with setuptools.

So it needs to be moved into `run`.